cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -480,6 +486,455 @@ var moveTests = []struct {
+			}
+
+			dir, err := ioutil.TempDir("/tmp", "cluster-api")
+			if err != nil {
+				t.Error(err)
+			}
+			defer os.RemoveAll(dir)
+
+			for _, node := range graph.uidToNode {
+				err = mover.backupTargetObject(node, dir)
+				if tt.wantErr {
+					g.Expect(err).To(HaveOccurred())
+					return
+				}
+
+				g.Expect(err).NotTo(HaveOccurred())
+
+				// objects are stored and serialized correctly in the temporary directory
+				expectedFilename := node.getFilename()
+				expectedFileContents, ok := tt.files[expectedFilename]
+				if !ok {
+					t.Errorf("Could not access file map: %v\n", expectedFilename)
+				}
+
+				path := filepath.Join(dir, expectedFilename)
+				fileContents, err := os.ReadFile(path)
+				if err != nil {
+					g.Expect(err).NotTo(HaveOccurred())
+					return
+				}
+
+				firstFileStat, err := os.Stat(path)
+				if err != nil {
+					g.Expect(err).NotTo(HaveOccurred())
+					return
+				}
+
+				fmt.Printf("Actual file content %v\n", string(fileContents))
+				g.Expect(string(fileContents)).To(Equal(expectedFileContents))
+
+				// Add delay so we ensure the file ModTime of updated files is different from old ones in the original files
+				time.Sleep(time.Millisecond * 5)
+
+				// Running backupTargetObject should override any existing files since it represents a new backup
+				err = mover.backupTargetObject(node, dir)
+				if tt.wantErr {
+					g.Expect(err).To(HaveOccurred())
+					return
+				}
+
+				g.Expect(err).NotTo(HaveOccurred())
+
+				secondFileStat, err := os.Stat(path)
+				if err != nil {
+					g.Expect(err).NotTo(HaveOccurred())
+					return
+				}
+
+				g.Expect(firstFileStat.ModTime()).To(BeTemporally("<", secondFileStat.ModTime()))
+			}
+		})
+	}
+}
+
+func Test_objectMover_restoreTargetObject(t *testing.T) {
+	// NB. we are testing the move and move sequence using the same set of moveTests, but checking the results at different stages of the move process
+	for _, tt := range backupRestoreTests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			// temporary directory
+			dir, err := ioutil.TempDir("/tmp", "cluster-api")
+			if err != nil {
+				g.Expect(err).NotTo(HaveOccurred())
+			}
+			defer os.RemoveAll(dir)
+
+			// Create an objectGraph bound a source cluster with all the CRDs for the types involved in the test.
+			graph := getObjectGraph()
+
+			// Get all the types to be considered for discovery
+			err = getFakeDiscoveryTypes(graph)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			// trigger discovery the content of the source cluster
+			g.Expect(graph.Discovery("")).To(Succeed())
+
+			// gets a fakeProxy to an empty cluster with all the required CRDs
+			toProxy := getFakeProxyWithCRDs()
+
+			// Run restoreTargetObject
+			mover := objectMover{
+				fromProxy: graph.proxy,
+			}
+
+			// Write go string slice to directory
+			for _, file := range tt.files {
+				tempFile, _ := ioutil.TempFile(dir, "obj")
+				tempFile.Write([]byte(file))
+				tempFile.Close()
```

Comment:
Please either explicitly handle or ignore the error (after a current PR from me is merged this will be a linter finding)